packages/flame_forge2d/lib/world_contact_listener.dart

```diff
@@ -0,0 +1,79 @@
+import 'flame_forge2d.dart';
+
+/// Listens to the entire [World] contacts events.
+///
+/// It propagates contact events ([beginContact], [endContact], [preSolve],
+/// [postSolve]) to other [ContactCallbacks]s when a [Body] or at least one of
+/// its fixtures `userData` is set to a [ContactCallbacks].
+///
+/// {@template flame_forge2d.world_contact_listner.algorithm}
+/// If the [Body] `userData` is set to a [ContactCallbacks] the contact events
+/// of this will be called to when any [Body]'s fixture contacts another
+/// [Fixture].
+///
+/// If instead you wish to be more specific and only trigger contact events
+/// when a specific [Body]'s fixture contacts with another [Fixture] you can
+/// set the fixture `userData` to a [ContactCallbacks].
+///
+/// If the colliding [Fixture] `userData` and [Body] `userData` are `null`, then
+/// the contact events are not called.
+///
+/// The described behaviour is a simple out of the box solution to propagate
+/// contact events. If you wish to implement your own logic you can subclass
+/// [ContactListener] and provide it to your [Forge2DGame].
+/// {@endtemplate}
+class WorldContactListener extends ContactListener {
+  void _callback(
+    Contact contact,
+    void Function(ContactCallbacks contactCallback, Object other) callback,
+  ) {
+    final userDatas = {
+      contact.bodyA.userData,
+      contact.fixtureA.userData,
+      contact.bodyB.userData,
+      contact.fixtureB.userData,
+    }.whereType<Object>();
+
+    for (final contactCallback in userDatas.whereType<ContactCallbacks>()) {
+      for (final object in userDatas) {
+        if (object != contactCallback) {
+          callback(contactCallback, object);
+        }
+      }
+    }
+  }
+
+  @override
+  void beginContact(Contact contact) {
+    _callback(
+      contact,
+      (contactCallback, other) => contactCallback.beginContact(other, contact),
```

Comment:
Hmm, I think that creating new lamdas is as expensive as creating new objects (not a 100% certain though) so would be good to avoid creating new lamdas here and on the other methods.

Reply:
What is your suggestion? Before we were doing ` _callbacks.forEach((c) => _maybeCallback(contact, c, c.begin));`. A solution would be to repeat the code, then it would be a compromise between readability, maintainability and performance. 

Let me know what you want to go forward with.

Reply:
Let's skip optimizing this for now and come back to it if need be.

Reply:
This would do it right? The method would be (if I remember correctly) be optimised to a constant value at compile time.

```dart

  @override
  void beginContact(Contact contact) {
    void passBeginCallback(contactCallback, other) => contactCallback.beginContact(other, contact),
  
    _callback(
      contact,
      passBeginContact,
    );
  }
```

Reply:
Ah yes, very good suggestion @wolfenrain! :clap:

Reply:
Hmm, but the method captures a local variable `contact`. Not sure how this works in Dart, but in C++ this would require making a new lambda object.

Reply:
True seems like it doesn't matter then, but according to the Flutter performance channel this shouldn't be expensive either so let's keep @alestiago's current solution and then optimize this if need be.